File: fast_sync_dart_client/fast_sync_client/lib/src/absraction/service/isync_manager.dart
import 'package:fast_sync_client/src/absraction/models/sync_payload.dart';

abstract class ISyncManager {
  Future<SyncPayload> push();
  Future<SyncPayload> pull();
  Future<SyncPayload> sync();
  Future<SyncPayload> hardReset({List<Type>? types});
}

File: fast-sync-typescript-client-sdk/packages/fast-sync-client/src/abstraction/service/isync_manager.ts
import { SyncPayload } from "../models/sync_payload";

export interface ISyncManager {
  push(): Promise<SyncPayload>;
  pull(): Promise<SyncPayload>;
  sync(): Promise<SyncPayload>;
  hardReset(types?: Array<new () => any>): Promise<SyncPayload>;
}
